### PR TITLE
NumberInput With ActualData

### PR DESCRIPTION
**Pull request by sarajaur**
# Related Issue

A Pull Request should close a **single** issue; multiple issues can be closed when the issues are **small and related**, but that should be an exception not the rule. Please keep Pull Requests small and targeted; large 'code drops' with dozens of files will be closed and asked to split into reviewable pieces. Reviews that need to be large due to dependencies will be
reviewed on a case-by-case basis.

Please use one of the well-known [github fixes keywords](https://help.github.com/en/articles/closing-issues-using-keywords) to reference
the issue fixed with this PR (eg Fixes #<github issue number>). If an issue doesn't yet exist please create one to aid
in issue tracking.

**NOTE**: For multiple issues resolved by this PR use the corresponding keywords **every time** in a comma-delimited list per the reference
page above.

# Description

For all Pull Requests, please describe how the issue was fixed or how the feature was implemented from a summary level. This information will be used to help provide context to the reviewers of the pull request and should be additive to the issue being closed.

- Implemented Number Input UI with Actual Data.

# Sample Card

If appropriate, please include a link to a card in one of the samples directories that can be used to validate this change. This can be an existing card or a card added with this PR.

# How Verified

How you verified the fix, including one or all of the following:
1. New unit tests that were added if any. If none were added please add a quick line explaining why not.
2. Existing relevant unit/regression tests that you ran
3. Manual scenario verification if any; ***Do include .gif's or screenshots of the testing you performed here if you think that it
will aid in code reviews or corresponding fixes on other platforms for eg.***

### CODE REVIEW

File: source/qml_v2/AdaptiveCardQmlEngine/AdaptiveCardQmlTypes.h [hunk 1]
```diff
@@ -23,5 +23,11 @@ namespace AdaptiveCardQmlEngine
         qmlRegisterType(QUrl("qrc:qml/CardConstants.qml"), "AdaptiveCardQmlEngine", 1, 0, "CardConstants");
 
         qmlRegisterType(QUrl("qrc:qml/TextBlockRender.qml"), "AdaptiveCardQmlEngine", 1, 0, "TextBlockRender");
+        qmlRegisterType(QUrl("qrc:qml/ImageRender.qml"), "AdaptiveCardQmlEngine", 1, 0, "ImageRender");
+        qmlRegisterType(QUrl("qrc:qml/RichTextBlockRender.qml"), "AdaptiveCardQmlEngine", 1, 0, "RichTextBlockRender");
+        qmlRegisterType(QUrl("qrc:qml/NumberInputRender.qml"), "AdaptiveCardQmlEngine", 1, 0, "NumberInputRender");
+        qmlRegisterType(QUrl("qrc:qml/InputFieldClearIcon.qml"), "AdaptiveCardQmlEngine", 1, 0, "InputFieldClearIcon");
```
Comment (soumykun) [inline source/qml_v2/AdaptiveCardQmlEngine/AdaptiveCardQmlTypes.h:29]: Why the old items ImageRender, RichTectBlockRneder ...are showing up as newly added.....these should already be there ?

File: source/qml_v2/AdaptiveCardQmlEngine/models/AdaptiveWarning.h [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    public:
+        AdaptiveWarning(Code code, const std::string& message);
+        Code GetStatusCode() const;
+        const std::string& GetReason() const;
+
+    private:
+        const Code m_code;
+        const std::string m_message;
+    };
+
```
Comment (soumykun) [inline source/qml_v2/AdaptiveCardQmlEngine/models/AdaptiveWarning.h:24]: Why we need this class and also why included under models, lets connect and discuss..

File: source/qml_v2/AdaptiveCardQmlEngine/models/NumberInputModel.cpp [hunk 3]
```diff
@@ -0,0 +1,72 @@
+#include "NumberInputModel.h"
+#include "SharedAdaptiveCard.h"
+#include <QDebug.h>
+#include "Utils.h"
+#include "MarkDownParser.h"
+
+NumberInputModel::NumberInputModel(std::shared_ptr<AdaptiveCards::NumberInput> numberInput, QObject* parent) :
+    QObject(parent), mInput(numberInput)
+
```
Comment (soumykun) [inline source/qml_v2/AdaptiveCardQmlEngine/models/NumberInputModel.cpp:9]: Please remove new line here.

File: source/qml_v2/AdaptiveCardQmlEngine/models/NumberInputModel.cpp [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        AdaptiveCardQmlEngine::AdaptiveCardContext::getInstance().addHeightEstimate(AdaptiveCardQmlEngine::AdaptiveCardContext::getInstance().getEstimatedTextHeight(mInput->GetLabel()));
+        mEscapedLabelString = QString::fromStdString(AdaptiveCardQmlEngine::Utils::getBackQuoteEscapedString(mInput->GetLabel()));
+    }
+    else
+    {
+        if (mInput->GetIsRequired())
+        {
+            AdaptiveCardQmlEngine::AdaptiveCardContext::getInstance().AddWarning(AdaptiveCardQmlEngine::AdaptiveWarning(AdaptiveCardQmlEngine::Code::RenderException, "isRequired is not supported without labels"));
+        }
+    }
```
Comment (soumykun) [inline source/qml_v2/AdaptiveCardQmlEngine/models/NumberInputModel.cpp:52]: Please check indentations.

File: source/qml_v2/AdaptiveCardQmlEngine/qml/CardConstants.qml [hunk 5]
```diff
@@ -1,6 +1,6 @@
-import "AdaptiveCardUtils.js" as AdaptiveCardUtils
+import "JSUtils/AdaptiveCardUtils.js" as AdaptiveCardUtils
 import QtQuick 2.15
-pragma Singleton
+//pragma Singleton
```
Comment (soumykun) [inline source/qml_v2/AdaptiveCardQmlEngine/qml/CardConstants.qml:4]: Lets connect on this !

File: source/qml_v2/AdaptiveCardQmlEngine/CMakeLists.txt [hunk 6]
```diff
@@ -3,7 +3,7 @@ cmake_minimum_required(VERSION 3.14)
 if(WIN32)
     project(AdaptiveCardQmlEngine LANGUAGES CXX)
 elseif(UNIX)
-    project(AdaptiveCardQmlEngine)
+project(AdaptiveCardQmlEngine)
 endif()
```
Comment (soumykun) [inline source/qml_v2/AdaptiveCardQmlEngine/CMakeLists.txt:7]: Indentation.

File: source/qml_v2/AdaptiveCardQmlEngine/models/CollectionItemModel.h [hunk 7]
```diff
@@ -2,16 +2,18 @@
...
 #include <QAbstractListModel>
 #include <SharedAdaptiveCard.h>
-
 #include <TextBlock.h>
 #include <RichTextBlock.h>
 #include <Image.h>
 #include <TextInput.h>
+#include "RichTextBlock.h"
+#include <NumberInput.h>
 #include "Enums.h"
```
Comment (soumykun) [inline source/qml_v2/AdaptiveCardQmlEngine/models/CollectionItemModel.h:11]: We should ideally use double quotes for include  here, as these files are part of our project and not external library.
Please keep a note of this, will clean it up in a separate PR.

File: source/qml_v2/AdaptiveCardQmlEngine/CMakeLists.txt (hunk 6 above)
Comment (sarajaur) [reply source/qml_v2/AdaptiveCardQmlEngine/CMakeLists.txt]: Done

**Review by soumykun — APPROVED** [review]

**Review by ramkrishnajana — APPROVED** [review]

**Pull request merged**